test/bh/attrs-test.js

```diff
@@ -0,0 +1,102 @@
+var fixtures = require('../fixtures')('bemhtml');
+var compile = fixtures.compile;
+
+describe('attrs()(×)', function() {
+  it('should return empty attrs', function() {
+    compile(function() {
+      block('button').def()(function() {
+        return typeof this.ctx.attrs;
+      });
+    })
+      .apply({ block: 'button' })
+      .should.equal('undefined');
+  });
+
+  it('should return attrs', function() {
+    compile(function() {
+      block('button').def()(function() {
+        return this.ctx.attrs.type;
+      });
+    })
+      .apply({ block: 'button', attrs: { type: 'button' } })
+      .should.equal('button');
+  });
+
+  xit('should set attrs', function() {
+    compile(function() {
+      block('checkbox').attrs()({
+        name: undefined,
+        type: 'button',
+        disabled: false,
+        hidden: true,
+        value: null
+      });
+    })
+      .apply({ block: 'checkbox' })
+      .should.equal('<div class="checkbox" type="button" hidden></div>');
+  });
+
+  it('should not override user attrs', function() {
+    compile(function() {
+      block('button').attrs()({
+        type: 'button',
+        disabled: true
+      });
+    })
+      .apply({
+        block: 'button',
+        attrs: {
+          type: 'link',
+          disabled: undefined,
+          name: 'button'
+        }
+      })
+      .should.equal('<div class="button" type="link" name="button"></div>');
+  });
+
+  xit('should not override later declarations', function() {
```

Comment:
It’s not possible in bem-xjst. This is achieved if use `this.extend(applyNext(), { ... })`.